specs/light_client/sync_protocol.md

```diff
@@ -25,150 +26,73 @@
-        return get_split_offset(
-            count,
-            SHARD_COUNT * committee_count,
-            validator_memory.shard_id * committee_count + (1 if end else 0),
-        )
-
-    actual_earlier_committee = maximal_earlier_committee[
-        0:get_offset(earlier_validator_count, True) - get_offset(earlier_validator_count, False)
-    ]
-    actual_later_committee = maximal_later_committee[
-        0:get_offset(later_validator_count, True) - get_offset(later_validator_count, False)
-    ]
-    def get_switchover_epoch(index):
-        return (
-            bytes_to_int(hash(validator_memory.earlier_period_data.seed + int_to_bytes(index, length=3))[0:8]) %
-            PERSISTENT_COMMITTEE_PERIOD
-        )
-
-    # Take not-yet-cycled-out validators from earlier committee and already-cycled-in validators from
-    # later committee; return a sorted list of the union of the two, deduplicated
-    return sorted(list(set(
-        [i for i in actual_earlier_committee if epoch % PERSISTENT_COMMITTEE_PERIOD < get_switchover_epoch(i)]
-        + [i for i in actual_later_committee if epoch % PERSISTENT_COMMITTEE_PERIOD >= get_switchover_epoch(i)]
-    )))
 ```
-
-Note that this method makes use of the fact that the committee for any given shard always starts and ends at the same validator index independently of the committee count (this is because the validator set is split into `SHARD_COUNT * committee_count` slices but the first slice of a shard is a multiple `committee_count * i`, so the start of the slice is `n * committee_count * i // (SHARD_COUNT * committee_count) = n * i // SHARD_COUNT`, using the slightly nontrivial algebraic identity `(x * a) // ab == x // b`).
+def compute_persistent_committee_at_slot(memory: ValidatorMemory,
+                                         epoch: Epoch) -> Sequence[Tuple[BLSPubkey, Gwei]]:
+    current_period = memory.finalized_header.slot // SLOTS_PER_EPOCH // EPOCHS_PER_SHARD_PERIOD
+    target_period = epoch // EPOCHS_PER_SHARD_PERIOD
+    if target_period == current_period + 1:
+        earlier_committee, later_committee = memory.current_committee, memory.next_committee
+    elif target_period == current_period:
+        earlier_committee, later_committee = memory.previous_committee, memory.current_committee
+    else:
+        raise Exception("Cannot compute for this slot")
+    o = []
+    for pub, aux in zip(earlier_committee.pubkeys, earlier_committee.compact_validators):
+        if epoch % EPOCHS_PER_SHARD_PERIOD < (aux >> 16) % EPOCHS_PER_SHARD_PERIOD:
+            o.append((pub, aux & (2**15-1))
+    for pub, aux in zip(later_committee.pubkeys, later_committee.compact_validators):
+        if epoch % EPOCHS_PER_SHARD_PERIOD >= (aux >> 16) % EPOCHS_PER_SHARD_PERIOD:
+            o.append((pub, aux & (2**15-1))
+    return o
+```
 
 ## Verifying blocks
 
 If a client wants to update its `finalized_header` it asks the network for a `BlockValidityProof`, which is simply:
```

Comment:
Might be useful to specify what parameters we pass to the network/server to request a particular proof